src/ralph/data_center/models/networks.py

```diff
@@ -0,0 +1,523 @@
+                    ip_range = []
+                ranges_and_networks.append((addr.min_ip, addr.max_ip, addr))
+
+        def f(a):
+            if a[2] == 0:
+                range_type = 'free'
+            elif a[2] == 1:
+                range_type = 'addr'
+            else:
+                range_type = a[2]
+            return {
+                'range_start': str(ipaddress.ip_address(a[0])),
+                'range_end': str(ipaddress.ip_address(a[1])),
+                'type': range_type,
+                'amount': a[1] - a[0] + 1,
+            }
+
+        min_ip = self.min_ip
+        parsed = []
+        for i, range_or_net in enumerate(ranges_and_networks):
+            if range_or_net[0] != min_ip:
+                parsed.append(f((min_ip, range_or_net[0] - 1, 0)))
+            parsed.append(f(range_or_net))
+            min_ip = range_or_net[1] + 1
+        if ranges_and_networks and ranges_and_networks[-1][1] < self.max_ip:
+            parsed.append(f((ranges_and_networks[-1][1] + 1, self.max_ip, 0)))
+        if not ranges_and_networks:
+            # this network is empty, lets append big, free block
+            parsed.append(f((self.min_ip, self.max_ip, 0)))
+        return parsed
+
+    def get_netmask(self):
+        try:
+            mask = self.address.split("/")[1]
+            return int(mask)
+        except (ValueError, IndexError):
+            return None
+
+
+class NetworkTerminator(NamedMixin, models.Model):
+
+    class Meta:
+        verbose_name = _('network terminator')
+        verbose_name_plural = _('network terminators')
+        ordering = ('name',)
+
+
+class DiscoveryQueue(NamedMixin, models.Model):
+
+    class Meta:
+        verbose_name = _('discovery queue')
+        verbose_name_plural = _('discovery queues')
+        ordering = ('name',)
+
+
+class IPAddress(LastSeenMixin, TimeStampMixin, models.Model):
+    asset = models.ForeignKey(
+        Asset, verbose_name=_('asset'), null=True, blank=True,
+        default=None, on_delete=models.SET_NULL,
+    )
+    address = models.GenericIPAddressField(
+        _('IP address'), help_text=_('Presented as string.'), unique=True,
+        blank=True, null=True, default=None,
+    )
+    number = models.BigIntegerField(
+        _('IP address'), help_text=_('Presented as int.'), editable=False,
+        unique=True,
+    )
+    hostname = models.CharField(
+        _('hostname'), max_length=255, null=True, blank=True, default=None,
+    )
+    # hostname.max_length vide /usr/include/bits/posix1_lim.h
+    is_management = models.BooleanField(
+        _('This is a management address'),
+        default=False,
+    )
+    network = models.ForeignKey(
+        Network, verbose_name=_('network'), null=True, blank=True,
+        default=None,
+    )
+    is_public = models.BooleanField(
+        _('This is a public address'),
+        default=False,
+        editable=False,
+    )
+    # TODO: old venture
```

Comment:
remove it. if we'll need anything like this it'd be service-env instead of venture.